requests/sessions.py

```diff
@@ -47,6 +48,16 @@
     preferred_clock = time.time
 
 
+class _RedirectedExceptionData(Exception):
```

Comment:
I understand this is all just an idea, but I'm curious why this inherits from Exception here.

Reply:
Just an artifact.  An earlier version was `_RedirectedException()` and was raised. After I got things working, I realized i should try to standardize behavior between an error in redirect chain (which requests detects) and an error in an immediately requested url (which urllib3 detects).  That meant a placeholder object (not raised) and the led to the idea of having a faux response-like object (which could be appropriate for exceptions raised in `adapters.py`.